scripts/generate_settings.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+import yaml
+import jinja2
+import re
+from settings_list import SettingsList
+import os
+
+sbp_generate_path = "../libswiftnav/sbp_generate"
+myenviron = os.environ
+texinputs = myenviron["TEXINPUTS"]
+myenviron["TEXINPUTS"]=".:" + sbp_generate_path + ":" + texinputs
+
+settings = SettingsList("settings.yaml")
+groups = settings.sl_return_groups()
+
+LATEX_SUBS = (
+    (re.compile(r'\\'), r'\\textbackslash'),
+    (re.compile(r'([{}_#%&$])'), r'\\\1'),
+    (re.compile(r'~'), r'\~{}'),
+    #(re.compile(r'\^'), r'\^{}'),
+    (re.compile(r'_'), r'_'),
+    (re.compile(r'"'), r"''"),
+    (re.compile(r'\.\.\.+'), r'\\ldots'),
+)
+
+NO_UNDERSCORE = re.compile(r'_')
+
+# We sometimes need to remove underscores.
+# This will remove the latex safe underscore character and replace with a space
+def no_us(value):
+    newval = value
+    if newval:
```

Comment:
Is this if in case newval is None or something that can't be subsituted? It's often a more Pythonic to do something like

```
try:
  return NO_UNDERSCORE.sub(' ', newval)
except SomeError:
  pass
return None
```